### PR TITLE
Adding email reporting functionality longevity test

### PR DESCRIPTION
**Pull request by aleksbykov**
- Move email sending functionality to base ClusterTester class
 - Refactoring email send functionality. remove the duplicate code

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by bentsi** [comment]
why it is draft?

**Comment by bentsi** [comment]
resolve conflicts

**Comment by aleksbykov** [comment]
@bentsi First was draft, because now appears new  thoughts how to improve

File: sdcm/cluster.py [hunk 1]
```diff
@@ -3173,10 +3173,7 @@ def node_run_stress_bench(node, loader_idx, stress_cmd, node_list):
             CassandraStressEvent(type='start', node=str(node), stress_cmd=stress_cmd)
 
             logdir = os.path.join(output_dir, self.name)
-            try:
-                os.makedirs(logdir)
-            except OSError:
-                pass
+            makedirs(logdir)
```
Comment (bentsi) [inline sdcm/cluster.py]: why did you remove try...except?

File: sdcm/tester.py [hunk 2]
```diff
@@ -256,6 +262,7 @@ def setUp(self):
         cluster.Setup.reuse_cluster(False)
         if self.monitors.nodes:
             self.prometheusDB = PrometheusDBStats(host=self.monitors.nodes[0].public_ip_address)
+        self.start_time = time.time()
```
Comment (bentsi) [inline sdcm/tester.py]: why do you init `self.start_time` twice?

File: sdcm/utils/common.py [hunk 3]
```diff
@@ -1071,3 +1071,16 @@ def get_tables_columns_list(entity_type):
...
                 avaialable_ks_cf.pop(ks_cf)
     return avaialable_ks_cf.keys()
+
+
+def format_timestamp(timestamp):
+    return datetime.datetime.utcfromtimestamp(timestamp).strftime('%Y-%m-%d %H:%M:%S')
+
+
+def makedirs(path):
+    try:
```
Comment (bentsi) [inline sdcm/utils/common.py]: what are you trying to solve here?

It raises an error exception if the leaf directory already exists or cannot be created. 
https://docs.python.org/2/library/os.html#os.makedirs

Comment (bentsi) [reply sdcm/utils/common.py]: There are 2 cases here:
- if it cannot be created there is nothing to do (the path won't exist)
- if it already exists, `makedirs` succeed <- maybe you wanted to solve this?

Comment (aleksbykov) [reply sdcm/utils/common.py]: The idea was to remove duplicating code in several places where several dirs have to be created. I wanted to put this in one place and emulate the functionality from python3 version of os.makedirs ( it has in python3 the new flag exists( if exists then do not throw exception) 
In out code, in several places, if folder is already exists, exception is thrown, and test/teardown or process is interrupted.

File: sdcm/cluster.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/cluster.py]: described below

File: sdcm/tester.py (hunk 2 above)
Comment (aleksbykov) [reply sdcm/tester.py]: it set twice for better count. First time it initiated at moment, when sct is starting. if setup will failed, report will have the time of start and there will be no exception of missing attribute. 
second time it is initiated when setup is successfully finished. At this moment we have more accurate time, when actually test was started

**Comment by aleksbykov** [comment]
@bentsi, i resolved latest conflicts and answered your comments

File: sdcm/utils/common.py (hunk 3 above)
Comment (bentsi) [reply sdcm/utils/common.py]: the problem is in the logic (generalization is OK):
Your code does:
tries to create directory
as I commented: 
if the leaf directory is created -> everything is OK.
if it wasnt' created exception is thrown.
Question: why do you check if the path exists when Python throws exception that the directory can't be created?
